### PR TITLE
[JENKINS-36720] Spotbugs fixes

### PR DESCRIPTION
**Pull request by res0nance**
Just more spotbugs fixes
See [JENKINS-36720](https://issues.jenkins-ci.org/browse/JENKINS-36720).

### Proposed changelog entries

* Entry 1: Issue, Human-readable Text
* ...

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] JIRA issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [x] There are at least 2 approvals for the pull request and no outstanding requests for change
- [x] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [x] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [x] Proper changelog labels are set so that the changelog can be generated automatically
- [x] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [x] If it would make sense to backport the change to LTS, a JIRA issue should exist and be labeled as `lts-candidate`

### CODE REVIEW

File: core/src/main/java/hudson/triggers/SCMTrigger.java [hunk 1]
```diff
@@ -557,7 +559,7 @@ public Runner(Action[] actions) {
             if (actions == null) {
                 additionalActions = new Action[0];
             } else {
-                additionalActions = actions;
+                additionalActions = Arrays.copyOf(actions, actions.length);
```
Comment (res0nance) [inline core/src/main/java/hudson/triggers/SCMTrigger.java]: This technically causes an issue where if a modification to `actions` occurs outside of the ctor this class would no longer see that effect. But i don't think we support that case

Comment (varyvol) [reply core/src/main/java/hudson/triggers/SCMTrigger.java]: It sounds to me like it would be a bug we need to fix if the situation arises.

**Review by varyvol — APPROVED** [review]

File: core/src/main/java/hudson/triggers/SCMTrigger.java (hunk 1 above)
Comment (alecharp) [reply core/src/main/java/hudson/triggers/SCMTrigger.java]: I'm not sure it would be a bug here but a "wrong usage" on the other side, no?

**Review by fcojfernandez — APPROVED** [review]
Looks good

**Review by alecharp — APPROVED** [review]

File: core/src/main/java/hudson/triggers/SCMTrigger.java (hunk 1 above)
Comment (res0nance) [reply core/src/main/java/hudson/triggers/SCMTrigger.java]: It should be a bug but I'm merely making a point of mentioning it.

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
Looks good, thanks for the continued work here.
Let's merge it after 24 hours, assuming no negative feedback,

thanks again 😺

**Review by StefanSpieker — APPROVED** [review]

**Pull request merged**